examples/brutus.jl

```diff
@@ -143,7 +143,11 @@ function code_mlir(f, types)
         for sidx in bb.stmts
             stmt = ir.stmts[sidx]
             inst = stmt[:inst]
-            line = ir.linetable[stmt[:line]+1]
+            line = @static if VERSION <= v"1.11"
+                ir.linetable[stmt[:line]+1]
+            else
+                (; ((:file, :line) .=> Base.IRShow.debuginfo_firstline(ir.debuginfo))...)
```

Comment:
IIUC, something like this should be more precise to each statement (though we don't really use locations anyway):
```suggestion
                last(Base.IRShow.buildLineInfoNode(ir.debuginfo, :var"n/a", sidx))
```
Feel free to choose either.

Reply:
buildLineInfoNode sometimes returns an empty list. I pushed a change that uses `debuginfo_firstline` as a fallback in that case, but maybe there's a cleaner solution...